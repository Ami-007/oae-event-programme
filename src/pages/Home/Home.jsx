import React from "react";
import "./Home.scss";
import ViewScheduleButton from "../../components/ViewScheduleButton/ViewScheduleButton";
import Paragraph from "../../components/Paragraph/Paragraph";
import EventHeaderDetails from "../../components/EventHeaderDetails/EventHeaderDetails";
import { useParams } from "react-router-dom";
import { getMockEventById } from "../../assets/data/data";
import YoutubeEmbed from "../../components/YoutubeEmbed/YoutubeEmbed";
import EventHeaderTitle from "../../components/EventHeaderTitle/EventHeaderTitle";

const Home = () => {
  const { eventId } = useParams();
  const { location, content } = getMockEventById(eventId);

  return (
    <div className="home">
      <h1 className="home__header">Home page</h1>
      <EventHeaderDetails
        time="11.30am"
        date="Sun 21 Nov 2021"
        venue="South Bank"
        location={location}
      />
      <Paragraph text={content[0].text} />
      <EventHeaderTitle title="BACH, THE UNIVERSE & EVERYTHING" subtitle="Can You Hear The Shape of The Drum? "/>
      <EventHeaderDetails time="11.30am" date="Sun 21 Nov 2021" venue="South Bank" location="London" />
      <YoutubeEmbed embedId="q6EoRBvdVPQ" />
      <Paragraph text="this is some paragraph" />
      <ViewScheduleButton />
     
    </div>
  );
};

export default Home;
