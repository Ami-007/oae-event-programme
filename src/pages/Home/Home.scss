@use "../../assets/sass/variables.scss" as *;
@use "../../assets/sass/typography.scss" as *;
.home {
  &__content {
    padding: 0 2.5rem;
  }
}
