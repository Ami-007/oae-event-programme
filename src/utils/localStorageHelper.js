export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
