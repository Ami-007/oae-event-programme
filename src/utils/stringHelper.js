export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
