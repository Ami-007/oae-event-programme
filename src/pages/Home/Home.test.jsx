import { customRender } from "../../utils/testUtils";
import Home from "./Home";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest.fn().mockReturnValue({ eventId: 1234 }),
  useRouteMatch: () => ({ url: "/oae-event-programme/1234" }),
}));

test("Test", () => {
  const { container } = customRender(<Home />);
  expect(container).toMatchSnapshot();
});
