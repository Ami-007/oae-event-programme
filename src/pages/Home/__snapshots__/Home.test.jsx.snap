// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
<div>
  <div
    class="home"
  >
    <h1
      class="home__header"
    >
      Home page
    </h1>
    <div
      class="header-home"
    >
      <h1
        class="header-home__title"
      >
        BACH, THE UNIVERSE & EVERYTHING
      </h1>
      <span
        class="header-home__subtitle"
      >
        Can You Hear The Shape of The Drum? 
      </span>
    </div>
    <div
      class="event-header-details"
    >
      <p
        class="event-header-details__time-date"
      >
        11.30am
        <span
          class="event-header-details__divider"
        >
          |
        </span>
        Sun 21 Nov 2021
      </p>
      <p
        class="event-header-details__venue-location"
      >
        <span
          class="event-header-details__venue"
        >
          South Bank
        </span>
        <span
          class="event-header-details__comma"
        >
          ,
        </span>
         
        London
      </p>
    </div>
    <p
      class="paragraph"
      data-testid="paragraph"
    >
      We like to think of the series as a community, similar to the one Bach enjoyed in Leipzig where he produced cantatas at an extraordinary rate, providing innovative music for the weekly services at the church where he worked from 1723 until his death in 1750. At that time, the congregation was as open to science and new ideas as it was united by faith. It is in the spirit that we come together for a thought-provoking cantata and a talk from a leading scientist. &lt;br&gt; Professor Jon Chapman from the University of Oxford invites us to ponder the fleeting moment when sound waves reach our ears before they disappear. Using some careful calculations, Jon suggests that we might even be able to hear the shape of a drum.
    </p>
    <div
      class="header-home"
    >
      <h1
        class="header-home__title"
      >
        BACH, THE UNIVERSE & EVERYTHING
      </h1>
      <span
        class="header-home__subtitle"
      >
        Can You Hear The Shape of The Drum? 
      </span>
    </div>
    <div
      class="event-header-details"
    >
      <p
        class="event-header-details__time-date"
      >
        11.30am
        <span
          class="event-header-details__divider"
        >
          |
        </span>
        Sun 21 Nov 2021
      </p>
      <p
        class="event-header-details__venue-location"
      >
        <span
          class="event-header-details__venue"
        >
          South Bank
        </span>
        <span
          class="event-header-details__comma"
        >
          ,
        </span>
         
        London
      </p>
    </div>
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowfullscreen=""
      class="youtube-embed"
      frameborder="0"
      src="https://www.youtube.com/embed/q6EoRBvdVPQ"
      title="Embedded youtube"
    />
    <p
      class="paragraph"
      data-testid="paragraph"
    >
      this is some paragraph
    </p>
    <a
      class="view-schedule-button"
      href="/event-schedule"
    >
      View Schedule
    </a>
  </div>
</div>
`;
